### PR TITLE
github actions: add build + lint check workflow for coral_web 

### PR DESCRIPTION
**Pull request by misspia-cohere**
Adds build and lint check workflow for `coral_web`

### CODE REVIEW

File: .github/workflows/interface_tests.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Interface Tests
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - src/interfaces/coral_web/**
```
Comment (misspia-cohere) [inline .github/workflows/interface_tests.yml:7]: [source](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore)

File: .github/workflows/interface_tests.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src/interfaces/coral_web
+
+steps:
+  - name: Checkout repo
+    uses: actions/checkout@v3
+  - name: Check coral_web is formatted
+    run: pnpm format
```
Comment (wujessica) [inline .github/workflows/interface_tests.yml:21]: before we can use `pnpm` do we need to install it or does it just work :o

Comment (misspia-cohere) [reply .github/workflows/interface_tests.yml:25]: added, good call 👍

File: .github/workflows/interface_tests.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src/interfaces/coral_web
+
+steps:
+  - name: Checkout repo
+    uses: actions/checkout@v3
+    with:
+      fetch-depth: 2
```
Comment (misspia-cohere) [inline .github/workflows/interface_tests.yml:21]: leaving this at the default value of `0` fetches the entire history from all branches. Instead, we just fetch the 2 most recent commits to save on time and resources

**Review by tianjing-li — APPROVED** [review]
Wow!

File: .github/actions/js-install/actions.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  using: composite
+  steps:
+    - uses: pnpm/action-setup@v2.1.0
+      with:
+        version: "9.0.6"
+
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: "js/.nvmrc"
```
Comment (wujessica) [inline .github/actions/js-install/actions.yml:14]: this should all be `coral_web` instead of `js` right? 😅

Comment (misspia-cohere) [reply .github/actions/js-install/actions.yml]: trueee-- corrected https://github.com/cohere-ai/cohere-toolkit/pull/81/commits/efa9edd8801dbb3b2c861c1d69b56f4effd6089a

File: .github/actions/js-install/actions.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    - uses: pnpm/action-setup@v2.1.0
+      with:
+        version: "9.0.6"
+
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: "src/interfaces/coral_web/.nvmrc"
+        cache: "pnpm"
+        cache-dependency-path: "js/pnpm-lock.yaml"
```
Comment (wujessica) [inline .github/actions/js-install/actions.yml:16]: i think this also needs to be updated

File: .github/actions/js-install/actions.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    - name: Install dependencies with pnpm
+      shell: bash
+      working-directory: js
```
Comment (wujessica) [inline .github/actions/js-install/actions.yml:20]: same with here?

File: .github/actions/js-install/actions.yml (hunk 5 above)
Comment (misspia-cohere) [reply .github/actions/js-install/actions.yml]: :')

**Review by wujessica — APPROVED** [review]
🚀

**Pull request merged**